### PR TITLE
Add CI & Documentation

### PR DESCRIPTION
**Pull request by rogerluan**
List of pending items:

- [x] Fix the preview/demo hyperlink
- [x] Add a demo gif
- [x] Prettify section [Usage outside of ActiveRecord](https://github.com/hoovbr/vin/pull/2#usage-outside-of-activerecord)
- [x] ~~Prettify section [Configuration](https://github.com/hoovbr/vin/pull/2#configuration)~~
- [x] Warn about sorting when timestamp is identical (same milisecond)
- [x] Fix CI

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Tests & Linter on PRs
+on: pull_request
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
```
Comment (matolah) [inline .github/workflows/pull-request.yml]: 😮

File: README.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+Follow the steps below to get started with VIN in your Ruby on Rails project. These steps assume your project is not yet live in production, so that you're free to make changes to your database schema and drop your existing database records.
+
+1. Make sure the primary key type is set to `:bigint` when generating new models
+
+To achieve this, create or update your `config/initializers/generators.rb` file:
+
+```
+Rails.application.config.generators do |g|
+  g.orm :active_record, primary_key_type: :bigint
+end
```
Comment (matolah) [inline README.md:82]: 🤩

File: README.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+```ruby
+…
+def set_vin_if_needed
+  # If using composite primary keys in Rails 7.1 and later
+  return if defined?(self.class.primary_key) && self.class.primary_key.is_a?(Array)
+  # If using composite primary keys in Rails 7.0 and earlier
+  return if defined?(self.class.primary_keys)
+  id_generator = Rails.application.config.id_generator
+  self.id ||= id_generator.generate_id(self.class::VIN_DATA_TYPE)
+end
```
Comment (matolah) [inline README.md:147]: 🚀

**Review by matolah — DISMISSED** [review]
What a README, it absolutely answered all the questions I had, great repo 👏👏👏👏

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (rogerluan) [reply .github/workflows/pull-request.yml]: I've removed this comment, it was a leftover :)

**Pull request merged**